dxlclient/broker.py

```diff
@@ -229,25 +230,42 @@ def _to_broker_string(self):
             "{}{}".format(self._FIELD_SEPARATOR, self._ip_address)
             if self.ip_address else "")
 
-    def _connect_to_broker(self):
+    def _connect_to_broker(self, **proxy):
         """
         Attempts to connect to a broker.
         Upon success will set the response time and whether the response
         came from the IP address versus the hostname.
+        :param proxy Proxy arguments dictionary(Can be empty)
         :return: None
         """
         broker_s = None
+        proxy_addr = proxy.get("proxy_addr", None)
+        proxy_port = proxy.get("proxy_port", None)
         try:
             start = datetime.datetime.now()
-            broker_s = socket.create_connection((self._host_name, self._port), timeout=1.0)
+            if proxy_addr is not None and proxy_port is not None:
+                logger.debug("Using proxy for connection: %s:%s", proxy_addr, proxy_port)
+                broker_s = socks.create_connection((self._host_name, self._port), timeout=1.0, **proxy)
+            else:
+                logger.debug("Not using proxy for connection.")
+                logger.debug("Please make sure Proxy section is added and UseWebSockets flag "
+                             "is set to True in client config file in order to use proxy.")
+                broker_s = socket.create_connection((self._host_name, self._port), timeout=1.0)
             end = datetime.datetime.now()
             self._response_from_ip_address = False
             self._response_time = (end - start).total_seconds()
         except socket.error as msg:
             if self._ip_address:
                 try:
                     start = datetime.datetime.now()
-                    broker_s = socket.create_connection((self._ip_address, self._port), timeout=1.0)
+                    if proxy_addr is not None and proxy_port is not None:
```

Comment:
Seems like it would be better to add a flag that indicates if proxy is enabled vs. checking for port and address in multiple places.